src/main.cc

```diff
@@ -12,14 +12,55 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <fstream>
 #include <iostream>
+#include <string>
 
 #include "class.h"
 
-int main(int argc, char *argv[]) {
+// Return true on success or false on failure.
+static bool ReadImage(const char filename[], WebPPicture* const pic) {
+  const uint8_t* data = NULL;
+  size_t data_size = 0;
+  WebPImageReader reader;
```

Comment:
You do not need to define all your variables at the beginning of the file: declare them whenever you need them: it makes the code shorter, it helps the compiler and prevents the person who codes to back back and forth.
E.g., below, please write: WebPImageReader reader = WebPGuessImageReader(data, data_size);